### PR TITLE
Exposed query as DBObject

### LINKED ISSUE #1842: Expose the QueryMetadata on the AbstractMongodbQuery so it can be converted to a Morphia Query
I use QueryDSL not just for querying, but also for deleting and updating mongo entities.  Since QueryDSL only support queries, I have to convert the AbstractMongodbQuery into a Morphia query (or break it down into a DBObject and obtain query modifiers to talk directly using the java driver).

The only workaround I know to convert a QueryDSL to a Morphia Query / DBObject is a hack whereby the query is serialized to JSON, deserialized to a DBObject and then jacked into a Morphia query using reflection to call an otherwise private method:

See https://groups.google.com/forum/#!topic/querydsl/ItCdp5wR6Vs

This works in simple cases where there is no limit or sorting on the query, but we lose modifier information.

Would it be possible to expose a getMetadata() on AbstractMongodbQuery so that we can get the limit, order and projection modifiers?

Alternatively, a "toMorphiaQuery" method on the AbstractMongodbQuery would also be a good addition.

I wouldn't mind contributing with a patch - just let me know which solution seems more appropriate.

Gaspard

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1842

### CODE REVIEW

File: querydsl-mongodb/src/main/java/com/querydsl/mongodb/AbstractMongodbQuery.java [hunk 1]
```diff
@@ -411,6 +411,15 @@ public void setReadPreference(ReadPreference readPreference) {
...
+     *
+     * @return
+     */
+    public DBObject asDBObject() {
+        return createQuery(queryMixin.getMetadata().getWhere());
+    }
+
     @Override
     public String toString() {
         return createQuery(queryMixin.getMetadata().getWhere()).toString();
```
Comment (Shredder121) [inline querydsl-mongodb/src/main/java/com/querydsl/mongodb/AbstractMongodbQuery.java]: Would we be able to reuse the `asDBObject` here in the `toString()`?
I don't know if they will ever diverge, but it reduces repetition right now.

Comment (timowest) [inline querydsl-mongodb/src/main/java/com/querydsl/mongodb/AbstractMongodbQuery.java]: good point

**Comment by johnktims** [comment]
Looks good to me.

**Comment by Shredder121** [comment]
To me as well.
I'll merge it then.

**Pull request merged**